src/core/ext/transport/chttp2/server/chttp2_server.cc

```diff
@@ -851,23 +851,27 @@ grpc_error_handle Chttp2ServerAddPort(Server* server, const char* addr,
                                       grpc_channel_args* args,
                                       Chttp2ServerArgsModifier args_modifier,
                                       int* port_num) {
-  if (strncmp(addr, "external:", 9) == 0) {
+  if (addr == nullptr) {
+    return GRPC_ERROR_CREATE_FROM_STATIC_STRING(
+        "Invalid address: addr cannot be a nullptr.");
+  }
+  if (strlen(addr) >= 9 && strncmp(addr, "external:", 9) == 0) {
```

Comment:
I don't think this `strlen()` check is needed.  If `addr` is shorter, `strncmp()` will stop at its null terminator.

Reply:
Removed